test-suite/lazyobject.cpp

```diff
@@ -193,6 +193,11 @@ void LazyObjectTest::testNotificationLoop() {
 
 #endif
 
+    // We have produced a ring of dependencies which we break here
```

Comment:
Should we use RAII to guarantee that the unregistering happens?

Reply:
Maybe?  If we don't get here, though, it means that the test suite already failed.  Ok for now, I guess, but feel free to open a PR.

Reply:
In general 👍

But, now that the PR is merged, is it worth the effort?!

Reply:
It could be someone's way to spend some time if they get bored and run out of series to watch :)